### PR TITLE
fix: warn when attempting to unset a global config without flag

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
warns when attempting to unset a global config without the `--global` flag because the output is confusing
### What issues does this PR fix or reference?
@W-13825061@

### CODE REVIEW

File: messages/config.unset.md [hunk 1]
```diff
@@ -27,3 +27,7 @@ You must provide one or more configuration variables to unset. Run "sf config li
 # didYouMean
 
 Did you mean %s?
+
+# unsetGlobalWarning
+
+The %s config is still set globally, unset it by using the --global flag.
```
Comment (jshackell-sfdc) [inline messages/config.unset.md:33]:
```suggestion
The %s config variable is still set globally, unset it by using the --global flag.
```

Comment (jshackell-sfdc) [reply messages/config.unset.md:33]: We use "config variable" in the docs: https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_dev_cli_config_values.htm

**Review by jshackell-sfdc — APPROVED** [review]
One small edit, but approving anyway 'cause I know you'll deal with it :)

File: src/commands/config/unset.ts [hunk 2]
```diff
@@ -36,13 +36,18 @@ export class UnSet extends SfCommand<SetOrUnsetConfigCommandResult> {
...
+    const config = await Config.create(Config.getDefaultOptions(flags.global));
 
     await config.read();
     for (const key of argv as string[]) {
       try {
         const resolvedName = this.configAggregator.getPropertyMeta(key)?.newKey ?? key;
         config.unset(resolvedName);
+
+        if (!flags.global && this.configAggregator.getLocation(resolvedName) === 'Global') {
+          // if the user wanted to unset a global value, and it's not in the global, warn them
```
Comment (shetzel) [inline src/commands/config/unset.ts:48]: This comment is confusing to me, particularly the "and it's not in the global" part.  What do you think about wording it this way?
```suggestion
          // If the config var is still set globally after an unset and the user didn't have the `--global` flag set, warn them.
```

**Review by shetzel — CHANGES_REQUESTED** [review]

**Review by shetzel — APPROVED** [review]

**Comment by shetzel** [comment]
QA:
✅ - unset local only
✅ - unset global only
✅ - unset local (with nothing set) with global still there.  warns about the global still being set.
✅ - unset local (with it set) with global still there.  warns about the global still being set.

**Pull request merged**